components/script/dom/document.rs

```diff
@@ -4696,6 +4817,46 @@ pub fn determine_policy_for_token(token: &str) -> Option<ReferrerPolicy> {
     }
 }
 
+// https://html.spec.whatwg.org/multipage/#dom-document-nameditem-filter
+fn document_nameditem_filter(name: &Atom, node: &Node) -> bool {
```

Comment:
is_document_named element and separately is_window_named_element since the logic is subtly different between them and it's important not to mix them, up, but yeah that's as good a place as any for it.